mailpoet/lib/Automation/Integrations/MailPoet/Actions/SendEmailAction.php

```diff
@@ -158,13 +160,30 @@ public function run(StepRunArgs $args): void {
       throw InvalidStateException::create()->withMessage(sprintf("Cannot schedule an email for subscriber ID '%s' because their status is '%s'.", $subscriberId, $subscriberStatus));
     }
 
+    $meta = $this->getNewsletterMeta($args);
     try {
-      $this->automationEmailScheduler->createSendingTask($newsletter, $subscriber);
+      $this->automationEmailScheduler->createSendingTask($newsletter, $subscriber, $meta);
     } catch (Throwable $e) {
       throw InvalidStateException::create()->withMessage('Could not create sending task.');
     }
   }
 
+  private function getNewsletterMeta(StepRunArgs $args): array {
+    if (!$this->automationHasAbandonedCartTrigger($args->getAutomation())) {
+      return [];
+    }
+
+    $productSubject = $args->getSinglePayloadByClass(ProductsPayload::class);
+    $cartProductIds = array_map(
+      function(\WC_Product $product) {
+        return $product->get_id();
+      },
+      $productSubject->getProducts()
+    );
+
+    return [AbandonedCart::TASK_META_NAME => $cartProductIds];
```

Comment:
I suppose this is how we're passing the product IDs today in the "legacy" abandoned cart emails, right? Then it makes sense, I guess. I think, ideally, automation emails could be aware of subjects/fields. But that sounds like a big refactor for the future.